docs/API-functions.md

```diff
@@ -2057,9 +2051,49 @@ will provide this information on its own.
 
 The following flags are available, as collected in `logging_api.h` public header:
 
-* `SRT_LOGF_DISABLE_TIME`: Do not provide the time in the header
-* `SRT_LOGF_DISABLE_THREADNAME`: Do not provide the thread name in the header
-* `SRT_LOGF_DISABLE_SEVERITY`: Do not provide severity information in the header
-* `SRT_LOGF_DISABLE_EOL`: Do not add the end-of-line character to the log line
+- `SRT_LOGF_DISABLE_TIME`: Do not provide the time in the header
+- `SRT_LOGF_DISABLE_THREADNAME`: Do not provide the thread name in the header
+- `SRT_LOGF_DISABLE_SEVERITY`: Do not provide severity information in the header
+- `SRT_LOGF_DISABLE_EOL`: Do not add the end-of-line character to the log line
+
+## Time Access
+
+The following set of functions is intended to retrieve timestamps from clock used by SRT.
+This timestamp MUST be used by sender if it wants to explicitely assign a timestamp
+to a certain packet. The sender can pass the timestamp in `MSGCTRL::srctime` of `srt_sendmsg2(..)`
+function together with the packet being submitted to SRT.
+The time provided MUST equal or exceed connection start time (`srt_connection_uptime(..)`)
+of the SRT socket passed to `srt_sendmsg2(..)`.
+
+The current time value as of SRT internal clock can be retrieved using `srt_time_now()` function.
```

Comment:
Consider placing the following information:

There are two known use cases of why you might want to use `srctime`:

1. SRT passthrough (for stream gateways). In this case you simply retrieve packets from an SRT source and pass them transparently to an SRT output (possibly re-encrypting). In that case, for every packet you read should also preserve the original value of `srctime` as obtained from `srt_recvmsg2`, and the original `srctime` for the same packet should be then passed for `srt_sendmsg2`. This could be used to avoid time jitter resulting from varying time difference between the time of receiving and sending the same packet.

2. Stable timing source. Your live streaming procedure, when spreading packets evenly into the stream in time, might predefine times for every single packet to keep time distances perfectly equal, or you believe that your signal input delivers packets at exact times that should be assigned to them - and you would like to preserve these times at the SRT receiving side and avoid a jitter that may result from varying time differences between the packet arrival and the moment when sending it over SRT. In cases like that you might do the following:

* When the packet arrival time counts, grab current time at this moment using `srt_time_now()`.
* When you wanted a precalculated packet time, use your private relative time counter reset to 0 when your first packet is ready, and specify `srctime` as a sum of this counter and the connection uptime obtained from `srt_connection_uptime()`. Although you still have to synchronize sending times with these predefined times, by explicitly specifying the source time you avoid the jitter resulting from a lost accuracy of waiting time and unfortunate thread scheduling.

Note that the clock used for `srctime` is a "steady clock" provided by system libraries and **it is not the clock expressing the current system time** (do not use functions like `ftime`, `time` or `gettimeofday` for it). This is a monotonic clock, that is, the base point in time when this time has a value of 0 is unchangeable during the whole lifetime of the application. An example of such clock is `CLOCK_MONOTONIC` used by `clock_gettime` function or `std::chrono::steady_clock` in the standard C++ library. To avoid problems and misunderstanding you should rely exclusively on time values provided by `srt_time_now()` and `srt_connection_uptime()` functions.